components/bitbucket/new-pipeline-event.js

```diff
@@ -0,0 +1,83 @@
+const _ = require("lodash");
+const common = require("./common");
+const { bitbucket } = common.props;
+
+const EVENT_SOURCE_NAME = "New Pipeline Event (Instant)";
+
+module.exports = {
+  ...common,
+  name: EVENT_SOURCE_NAME,
+  description: "Emits an event when a pipeline event occurs",
+  version: "0.0.1",
+  props: {
+    ...common.props,
+    repositoryId: {
+      optional: true,
+      propDefinition: [
+        bitbucket,
+        "repositoryId",
+        c => ({ workspaceId: c.workspaceId }),
+      ],
+    },
+    eventTypes: {
+      type: "string[]",
+      label: "Pipeline Event Types",
+      description: "The type of pipeline events that will trigger this event source",
+      optional: true,
+      options: [
+        // See https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/
+        { label: 'Build started', value: 'INPROGRESS' },
+        { label: 'Build succeeded', value: 'SUCCESSFUL' },
+        { label: 'Build failed', value: 'FAILED' },
+      ],
+    },
+  },
+  methods: {
+    ...common.methods,
+    getEventSourceName() {
+      return EVENT_SOURCE_NAME;
+    },
+    getHookEvents() {
+      return [
+        'repo:commit_status_created',
+        'repo:commit_status_updated',
+      ];
+    },
+    getHookPathProps() {
+      return {
+        workspaceId: this.workspaceId,
+        repositoryId: this.repositoryId,
+      };
+    },
+    isEventRelevant(event) {
+      const { state: eventType } = event.body.commit_status;
+      return (
+        _.isEmpty(this.eventTypes) ||
+        this.eventTypes.some(i => i === eventType)
+      );
+    },
+    generateMeta(data) {
+      const {
+        "x-request-uuid": id,
+        "x-event-time": eventDate,
+      } = data.headers;
+      const {
+        repository,
+        state: eventType,
+      } = data.body.commit_status;
+      const summary = `New pipeline event in ${repository.name}: ${eventType}`;
+      const ts = +new Date(eventDate);
+      return {
+        id,
+        summary,
+        ts,
+      };
+    },
+    processEvent(event) {
+      if (this.isEventRelevant(event)) {
+        const parent = common.methods.processEvent.bind(this);
```

Comment:
interesting pattern with spreading common and this... we've discussed mixins / inheritance in the past (clearly code reuse has not been a priority) -- will be nice to take a look if certain strategies emerge

Reply:
There are some rough edges, like when trying to call a "parent" method (e.g. those defined in the `.app.js` file) without having the parent instance available.
So to sum up, I like that we ran into the same issues :slightly_smiling_face: